frontend/src/metabase/dashboard/components/DashboardTabs/use-dashboard-tabs.ts

```diff
@@ -13,24 +13,27 @@ import {
   moveTab as moveTabAction,
 } from "metabase/dashboard/actions";
 import { SelectedTabId } from "metabase-types/store";
-import { getDashboardId, getSelectedTabId } from "metabase/dashboard/selectors";
+import { getSelectedTabId, getTabs } from "metabase/dashboard/selectors";
 import { addUndo } from "metabase/redux/undo";
 
+import { useSyncURLSlug } from "./use-sync-url-slug";
+
 let tabDeletionId = 1;
 
-export function useDashboardTabs() {
+export function useDashboardTabs({
+  slug,
+  pathname,
+}: {
+  slug: string | undefined;
+  pathname: string;
+}) {
   const dispatch = useDispatch();
-  const dashboardId = useSelector(getDashboardId);
-  const tabs = useSelector(state =>
-    dashboardId
-      ? state.dashboard.dashboards[dashboardId].ordered_tabs?.filter(
-          tab => !tab.isRemoved,
-        ) ?? []
-      : [],
-  );
+  const tabs = useSelector(getTabs);
   const selectedTabId = useSelector(getSelectedTabId);
 
-  useMount(() => dispatch(initTabs()));
+  useSyncURLSlug({ slug, pathname });
+  useMount(() => dispatch(initTabs({ slug })));
+  useUnmount(() => dispatch(selectTab({ tabId: null })));
```

Comment:
In `useSyncURLSlug`, we check if `selectedTabId` was previously `null` to determine if a dashboard is being rendered for the first time. We need to clear `selectedTabId` when unmounting to ensure this is correct.